prospero-cli/src/main/java/org/wildfly/prospero/cli/RepositoryDefinition.java

```diff
@@ -32,90 +31,67 @@
             final String repoInfo = repos.get(i);
             final String repoId;
             final String repoUri;
 
-            try {
-                if (repoInfo.contains("::")) {
-                    final String[] parts = repoInfo.split("::");
-                    if (parts.length != 2 || parts[0].isEmpty() || parts[1].isEmpty()) {
-                        throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
-                    }
-                    repoId = parts[0];
-                    repoUri = parseRepositoryLocation(parts[1]);
-                } else {
-                    repoId = "temp-repo-" + i;
-                    repoUri = parseRepositoryLocation(repoInfo);
+            if (repoInfo.contains("::")) {
+                final String[] parts = repoInfo.split("::");
+                if (parts.length != 2 || parts[0].isEmpty() || parts[1].isEmpty()) {
+                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
                 }
-                repositories.add(new Repository(repoId, repoUri));
-            } catch (URISyntaxException e) {
-                logger.error("Unable to parse repository uri + " + repoInfo, e);
-                throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
+                repoId = parts[0];
+                repoUri = parseRepositoryLocation(parts[1]);
+            } else {
+                if (StringUtils.isBlank(repoInfo)) {
+                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
+                }
+                repoId = "temp-repo-" + i;
+                repoUri = parseRepositoryLocation(repoInfo);
             }
 
+            repositories.add(new Repository(repoId, repoUri));
         }
         return repositories;
     }
 
-    private static String parseRepositoryLocation(String repoLocation) throws URISyntaxException, ArgumentParsingException {
-        if (!isRemoteUrl(repoLocation) && !repoLocation.isEmpty()) {
-            // the repoLocation contains either a file URI or a path
-            // we need to convert it to a valid file IR
-            repoLocation = getAbsoluteFileURI(repoLocation).toString();
-        }
-        if (!isValidUrl(repoLocation)){
-            throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoLocation);
-        }
-        return repoLocation;
-    }
-
-    private static boolean isRemoteUrl(String repoInfo) {
-        return repoInfo.startsWith("http://") || repoInfo.startsWith("https://");
-    }
-
-    private static boolean isValidUrl(String text) {
+    private static String parseRepositoryLocation(String location) throws ArgumentParsingException {
+        URI uri;
         try {
-            new URL(text);
-            return true;
-        } catch (MalformedURLException e) {
-            return false;
+            uri = new URI(location);
+            if (("file".equals(uri.getScheme()) || StringUtils.isBlank(uri.getScheme()))
```

Comment:
I think there are some legal paths that would not be caught by this. For example, on Windows  -  `D:/Foo/Bar`, on Linux `a:foo/bar`. In both cases the first letter is treated as a schema and so the whole path will be treated as a URI and fail on `uri.toURL()`

Reply:
Technically I think Windows `D:/Foo/Bar` should not be allowed because it's not a URL, but it should be handled gracefully by the `catch (URISyntaxException e)` block bellow.

On Linux, `a:foo/bar` can as well be `http:foo/bar` and now how do we tell if it should be treated as URL or a path?

This is a hole we dug ourselves by being too benevolent about the input, we should have insisted on a URL :).